sycl/unittests/Extensions/DefaultContext.cpp

```diff
@@ -51,13 +68,31 @@ TEST(DefaultContextTest, DefaultContextCanBeDisabled) {
   sycl::unittest::PiMock Mock;
   sycl::platform Plt = Mock.getPlatform();
 
-  bool catchException = false;
-  try {
-    (void)Plt.ext_oneapi_get_default_context();
-  } catch (const std::runtime_error &) {
-    catchException = true;
+  test_default_context_disabled(Plt);
+}
+
+TEST(DefaultContextTest, DefaultContextCanBeDisabledEnabled) {
+  using namespace sycl::detail;
+  using namespace sycl::unittest;
+  {
+    sycl::unittest::PiMock Mock;
+    sycl::platform Plt = Mock.getPlatform();
+    Plt.detail_enable_ext_oneapi_default_context(false);
+
+    test_default_context_disabled(Plt);
   }
 
-  ASSERT_TRUE(catchException)
-      << "ext_oneapi_get_default_context did not throw and exception";
+  {
+    sycl::unittest::PiMock Mock1;
+    sycl::platform Plt1 = Mock1.getPlatform();
+
+    sycl::unittest::PiMock Mock2;
+    sycl::platform Plt2 = Mock2.getPlatform();
+
+    // Since the platforms were gotten by the same way (same selector)
+    // it should be sufficient to enable the extension for one of them
+    Plt1.detail_enable_ext_oneapi_default_context(true);
+
+    test_contexts_are_equal(Plt1, Plt2);
```

Comment:
I think we're missing a check to verify that default context is actually enabled.

Reply:
If it's the case then it was missed previously as well. As I wrote above I didn't change the logic of the test.

If you still think something extra should be there, please specify your concern more precisely because right now I don't quite understand what it is. Thanks.